presto-main/src/main/java/com/facebook/presto/sql/planner/QueryPlanner.java

```diff
@@ -447,6 +454,7 @@ private PlanBuilder aggregate(PlanBuilder subPlan, QuerySpecification node)
         Optional<Symbol> groupIdSymbol = Optional.empty();
         if (groupingSets.size() > 1) {
             groupIdSymbol = Optional.of(symbolAllocator.newSymbol("groupId", BIGINT));
+            symbolAllocator.addGroupIdSymbolForQuerySpecification(node, groupIdSymbol.get());
```

Comment:
You're right, it's one `groupId` symbol per `QuerySpecification`. So technically this `groupId` symbol should have a lifetime limited to the `plan(QuerySpecification)` method. I tried doing some refactoring to get there but couldn't find a more elegant solution than sticking the `groupId` symbol in the `SymbolAllocator`.

1. Tried adding a field to `PlanBuilder` that would keep track of the `groupId` symbol. That opened a can of worms. I had to make sure that every time a new `PlanBuilder` was built, it would correctly carry over the `groupId` symbol. Similarly for `TranslationMap`.
2. Return a `Pair<PlanBuilder, Symbol>` from the `aggregate(PlanBuilder, QuerySpecification)` method and then weave the returned symbol to the remaining methods. This seems ugly and breaks the symmetry of the method signatures. (This is the method I prefer the most).
3. Adding a map as a field to `QueryPlanner` that keeps track of which `groupId` symbol corresponds to which `QuerySpecification`.

What do you think? For now I'm leaving the code as is.